src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/endpoint/UserEndpoint.java

```diff
@@ -304,4 +313,52 @@ public UserEndpoint(
 
     return ResponseEntity.ok(userService.removeAdminRole(userIdentity, roleName));
   }
+
+  @GetMapping("/audit")
+  public ResponseEntity<List<UserAuditDTO>> audit(
+      @RequestParam(required = false) String performedBy,
+      @RequestParam(required = false) String performedOn)
+      throws CTPException {
+
+    log.info(
+        "Entering audit search", kv("performedBy", performedBy), kv("performedOn", performedOn));
+
+    // Verify that the caller can retrieve audit history
+    rbacService.assertUserPermission(PermissionType.READ_USER_AUDIT);
+
+    // Search the audit table
+    List<UserAudit> auditHistory;
+    if (performedBy != null && Strings.isNullOrEmpty(performedOn)) {
+      auditHistory = userAuditService.getAuditHistoryForPerformedBy(performedBy);
+    } else if (performedOn != null && Strings.isNullOrEmpty(performedBy)) {
+      auditHistory = userAuditService.getAuditHistoryForPerformedOn(performedOn);
+    } else {
+      throw new CTPException(
+          Fault.BAD_REQUEST, "Only one of 'performedByUser' or 'performedOnUser' must be supplied");
```

Comment:
should be Only one of  'performedBy' or 'performedOn'.  (ie drop the "User" bit)

or "Only one of 'perfomedBy=user' or 'performedOn=user' must be supplied"

Reply:
Fixed